python/arcticdb/storage_fixtures/mongo.py

```diff
@@ -0,0 +1,145 @@
+        """
+        Parameters
+        ----------
+        mongo_uri
+            URI to the MongoDB server, must start with "mongodb://".
+        name
+            The database name to use in the MongoDB server. Must not contain ``?`` symbol.
+            If not supplied, a random name is generated.
+            Regardless of whether the database exists before, it will be removed on ``__exit__`` of this Fixture.
+        client
+            Optionally reusing a client already connected to ``mongo_uri``.
+        """
+        super().__init__()
+        from pymongo import MongoClient
+
+        assert mongo_uri.startswith("mongodb://")
+        self.mongo_uri = mongo_uri
+        self.client = client or MongoClient(mongo_uri)
+        if not name:
+            while True:
+                name = f"MongoFixture{time.time()}"
+                if name not in self.client.list_database_names():
+                    break
+                time.sleep(0.01)
+        self.prefix = name + "."
+
+        PrefixingLibraryAdapterDecorator.ensure_registered()
+        self.arctic_uri = PrefixingLibraryAdapterDecorator.prefix_uri(self.prefix, mongo_uri)
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.libs_from_factory.clear()
+
+        with handle_cleanup_exception(self, "prefix_mongo_database"):
+            self.client.drop_database("arcticc_" + self.prefix[:-1])
+
+        # With Mongo, the LibraryManager configuration database is global/reused across fixtures, so must delete the
+        # library definitions
+        self.slow_cleanup()
+
+    def create_test_cfg(self, lib_name: str) -> EnvironmentConfigsMap:
+        cfg = EnvironmentConfigsMap()
+        add_mongo_library_to_env(cfg, lib_name=lib_name, env_name=Defaults.ENV, uri=self.mongo_uri)
+        return cfg
+
+    def set_permission(self, *, read: bool, write: bool):
+        raise NotImplementedError("Will support setting permissions on Mongo soon")  # TODO
+
+
+class ExternalMongoDBServer(StorageFixtureFactory):
+    """A MongoDB server whose life-cycle is managed externally to this test fixture system."""
+
+    def __init__(self, mongo_uri: str):
+        self.mongo_uri = mongo_uri
+
+    def __str__(self):
+        return f"{type(self).__name__}[{self.mongo_uri}]"
+
+    def create_fixture(self) -> StorageFixture:
+        return MongoDatabase(self.mongo_uri)
+
+
+class ManagedMongoDBServer(StorageFixtureFactory):
+    """Represents a MongoDB server started by this class"""
+
+    _count = -1
+
+    def __init__(self, data_dir: Optional[str] = None, port=0, executable="mongod"):
+        self._data_dir = data_dir or tempfile.mkdtemp("ManagedMongoDBServer")
+        self._port = port or get_ephemeral_port()
+        self._executable = executable
+
+    def _safe_enter(self):
+        from pymongo import MongoClient
+
+        cmd = [self._executable, "--port", str(self._port), "--dbpath", self._data_dir, "--nojournal"]
```

Comment:
Do we really need `--nojournal`? It's removed in modern Mongos, and is in fact always on with them, which caused me a bit of pain when setting up the new dev machine.